### PR TITLE
Update DoublyStridedOpInterface to include both static and dynamic access attributes

### PR DESCRIPTION
**Pull request by jtuyls**
Addresses: https://github.com/nod-ai/iree-amd-aie/pull/267#r1563241598

### CODE REVIEW

**Comment by jtuyls** [comment]
@MaheshRavishankar Do you want to review this PR?

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEDmaOpInterface.td [hunk 1]
```diff
@@ -25,70 +25,233 @@ def DoublyStridedOpInterface : OpInterface<"DoublyStridedInterface"> {
...
+      /*methodBody=*/"",
+      /*defaultImplementation=*/[{
+        return $_op.getTargetStaticStrides();
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/[{
+        Return a vector of all the static or dynamic target offsets of the op.
+      }],
+      /*retTy=*/"::llvm::SmallVector<::mlir::OpFoldResult, 4>",
```
Comment (MaheshRavishankar) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEDmaOpInterface.td:109]: Nit: drop the `, 4` . What ends up happening is if you do something like

```
SmallVector<OpFoldResult, 5> ... = getTargetMixedOffsets();
```

then the move methods dont kick in and it creates a copy. So easiest is to just drop the size template parameter.

**Review by MaheshRavishankar — APPROVED** [review]
Oh nice idea having a new interface for this! Cool!

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEDmaOpInterface.td (hunk 1 above)
Comment (jtuyls) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEDmaOpInterface.td]: Cool, thanks for pointing this out! Updated the PR.

**Pull request merged**